### PR TITLE
Make pyaudio optional

### PR DESCRIPTION
**Pull request by dreinsch**
The goal of this PR is to make the pyaudio dependency optional (as extra pyaudio).

### CODE REVIEW

File: pya/helper/helpers.py [hunk 1]
```diff
@@ -111,9 +110,21 @@ def buf_to_float(x, n_bytes=2, dtype=np.float32):
...
+        )
+        raise RuntimeError(msg) from e
+    else:
+        return pyaudio.PyAudio()
+
+
 def device_info():
     """Return a formatted string about available audio devices and their info"""
-    pa = pyaudio.PyAudio()
+    pa = _try_importing_pyaudio("device_info")
```
Comment (thomas-hermann) [inline pya/helper/helpers.py]: as the return value is a `pyaudio.PyAudio()`, I suggest to change the function name to `_try_opening_pyaudio(fun_name)`

File: pya/arecorder.py [hunk 2]
```diff
@@ -108,7 +107,7 @@ def _recorder_callback(self, in_data, frame_count, time_info, flag):
             self.record_buffer.append(data_float)
             # E = 10 * np.log10(np.mean(data_float ** 2)) # energy in dB
             # os.write(1, f"\r{E}    | {self.block_cnt}".encode())
-        return None, pyaudio.paContinue
+        return self.backend.process_buffer(None)
```
Comment (thomas-hermann) [inline pya/arecorder.py:110]: ok, maybe add a comment that `process_buffer` is called both for input and output streams

**Review by thomas-hermann — COMMENTED** [review]
I generally approve the move. I added 2 minor comments in the code.
As Jiajun has a deeper look into this topic, he should approve and merge.

File: tests/helpers.py [hunk 3]
```diff
@@ -9,3 +10,19 @@ def wait(condition: Callable[[], bool], seconds: float = 10, interval: float = 0
...
+        import pyaudio
+        pyaudio.PyAudio().get_default_input_device_info()
+        return True
+    return False
+
+
+def check_for_output() -> bool:
+    with contextlib.suppress(ImportError, OSError):
+        import pyaudio
+        pyaudio.PyAudio().get_default_input_device_info()
```
Comment (wiccy46) [inline tests/helpers.py]: This should be output device info.

**Review by wiccy46 — APPROVED** [review]
Nice bit of change. Look sane to me apart from a minor issue. Looking forward to trying this out

**Pull request merged**